### PR TITLE
Enable fetching default gateway IP

### PR DESCRIPTION
**Pull request by parrotmac**
Enable fetching the default gateway of the Wi-Fi connection on iOS and Android.

~**Note:** Android version has been tested (Nexus 6P w/ 8.1.0), but iOS version is totally untested.~
**Update:**
Tested on:
- Nexus 6P w/ Android 8.1.0
- iPhone 8 w/ iOS 12.0

### CODE REVIEW

File: ios/RNNetworkInfo.m [hunk 1]
```diff
@@ -148,6 +149,22 @@ @implementation RNNetworkInfo
...
 
+RCT_EXPORT_METHOD(getGatewayIPAddress:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    @try{
+	NSString *ipString = nil;
+	struct in_addr gatewayaddr;
+	int r = getdefaultgateway(&(gatewayaddr.s_addr));
+	if(r >= 0) {
+        	ipString = [NSString stringWithFormat: @"%s",inet_ntoa(gatewayaddr)];
```
Comment (zacatac) [inline ios/RNNetworkInfo.m]: I'm surprised Java doesn't have something like `inet_ntoa`.

File: ios/RNNetworkInfo.m [hunk 2]
```diff
@@ -148,6 +149,22 @@ @implementation RNNetworkInfo
...
+    	}
+        resolve(gatewayaddr);
```
Comment (zacatac) [inline ios/RNNetworkInfo.m]: Don't we want to resolve the `ipString`?

**Review by zacatac — APPROVED** [review]
That was fast!

I think this is great for our use case, but I don't know if the maintainers will want to merge it upstream without some unit tests.

File: ios/RNNetworkInfo.m (hunk 1 above)
Comment (parrotmac) [reply ios/RNNetworkInfo.m]: It might, but then Android might not support it

File: ios/RNNetworkInfo.m (hunk 2 above)
Comment (parrotmac) [reply ios/RNNetworkInfo.m]: Good catch!

**Comment by parrotmac** [comment]
> I think this is great for our use case, but I don't know if the maintainers will want to merge it upstream without some unit tests.

@zacatac Any ideas on writing tests for these?

File: ios/getgateway.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//
+//  getgateway.h
+//  OBDInterface
```
Comment (parrotmac) [inline ios/getgateway.h]:
```
```

**Pull request merged**